app/src/main/java/io/neurolab/activities/ShareDataActivity.java

```diff
@@ -0,0 +1,160 @@
+    private static Set<File> selectedFiles = new HashSet<>();
+    private static Uri fileUri;
+    private static ArrayList<Uri> selectedUri = new ArrayList<Uri>();
+    private static Set<Uri> set;
+    private static SparseBooleanArray sparseBooleanArray;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_data);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        Toast.makeText(this, R.string.share_screen_toast, Toast.LENGTH_SHORT).show();
+
+        fileListView = findViewById(R.id.fileListView);
+        context = getApplicationContext();
+
+        resultIntent = new Intent(Intent.ACTION_SEND_MULTIPLE);
+        resultIntent.setType("text/plain");
+
+        StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
+        StrictMode.setVmPolicy(builder.build());
+
+        File appDir = new File(Environment.getExternalStorageDirectory().getAbsolutePath() + File.separator +
+                CSV_DIRECTORY);
+        File[] files = appDir.listFiles();
+        fileList = new String[files.length];
+        newFileName = new String[files.length];
+
+        if (appDir.listFiles() != null && files.length > 0) {
+            int j = 0;
+            for (int i = 0; i < files.length; i++) {
+                fileList[j] = files[i].getAbsolutePath();
+                newFileName[i] = files[i].getName();
+                j++;
+            }
+        }
+
+        final List<String> file_list = new ArrayList<String>(Arrays.asList(newFileName));
+        final ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>
+                (this, android.R.layout.simple_list_item_multiple_choice, file_list);
+        fileListView.setChoiceMode(ListView.CHOICE_MODE_MULTIPLE);
+        fileListView.setItemsCanFocus(false);
+        fileListView.setAdapter(arrayAdapter);
+
+        fileListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+        @Override
+        public void onItemClick(AdapterView<?> adapterView, View view, int position, long rowId) {
+
+           sparseBooleanArray = fileListView.getCheckedItemPositions();
+           File requestFile = new File(fileList[position]);
+
+           try {
+               fileUri = FileProvider.getUriForFile(ShareDataActivity.this, "io.neurolab.fileprovider", requestFile);
+           } catch (IllegalArgumentException e) {
+               e.printStackTrace();
+           }
+
+           for (int i = 0; i < fileListView.getCount(); i++) {
+               if (sparseBooleanArray.get(i)) {
+                   selectedFiles.add(files[i]);
+               } else if (!sparseBooleanArray.get(i)) {
+                   selectedFiles.remove(files[i]);
+               }
+           }
+          updateSelectedFiles(selectedFiles);
+          setPermission();
+         }
+       }
+      );}
+
+      private void setPermission()
+      {
```

Comment:
Reformat code

Reply:
@CloudyPadmal I have changed according to review.
Can you please check?